### ISSUE TITLE
 Running benchmark tests in Go fails

### ISSUE THREAD

**Issue by CocoaPriest** [zed-industries/zed#12693]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Running benchmark tests in Go fails. Regular tests work as expected.
_Hint_: I think it may be related to the fact I'm using **Fish shell**.

```
fish: Expected a variable name after this $.
go test $ZED_CUSTOM_GO_PACKAGE -benchmem -run=^$ -bench ^$ZED_SYMBOL$
                                               ^

⏵ Task `go test ./iteration -bench BenchmarkRepeat` finished with non-zero error code: 127
⏵ Command: '/usr/local/bin/fish -i -c `go test ./iteration -benchmem -run=^$ -bench ^BenchmarkRepeat$`'
```

### Environment

Zed: v0.138.4 (Zed)
OS: macOS 14.4.1
Memory: 32 GiB
Architecture: aarch64

**Comment by osiewicz** [comment]
Yep, that's just due to fish being incompatible with that syntax.

**Comment by notpeter** [comment]
The issue is here:
https://github.com/zed-industries/zed/blob/7c72929f0ba3ade5fde3e7b090dcd2685ae243e3/crates/languages/src/go.rs#L533-L539

I wonder whether it would be sufficient to replace `-run=^$"` with `-run='^$'"` (untested).
